fabric/dataclient.go

```diff
@@ -703,7 +492,281 @@ func stringToEmptyInterface(a []string) []interface{} {
+	if m.Ratelimit != nil {
+		r.Filters = append(r.Filters,
+			&eskip.Filter{
+				//inlineContentIfStatus(429, "{\"title\": \"Rate limit exceeded\", \"detail\": \"See the retry-after header for how many seconds to wait before retrying.\", \"status\": 429}", "application/problem+json")
+				Name: filters.InlineContentIfStatusName,
+				Args: []interface{}{
+					429,
+					"{\"title\":\"Rate limit exceeded\",\"detail\":\"See the retry-after header for how many seconds to wait before retrying.\",\"status\":429}",
+					"application/problem+json",
+				},
+			},
+			&eskip.Filter{
+				// clusterClientRatelimit("spp-brand-service_api-brand-assignments-id_DELETE", 30, "1m", "Authorization")
+				Name: filters.ClusterClientRatelimitName,
+				Args: []interface{}{
+					// TODO(sszuecs): maybe we want to add namespace here, too (assume people could use namespaces to separate prod/staging, this would otherwise count both)
+					fmt.Sprintf("%s_%s_%s",
+						name,
+						strings.Trim(nonWord.ReplaceAllString(path, "-"), "-"),
+						m.Method,
+					),
+					m.Ratelimit.DefaultRate,
+					m.Ratelimit.Period,
+					// optional header, TODO(sszuecs): maybe configurable in the future
+					"Authorization",
+				},
+			},
+		)
+	}
+
+	// add rest filters
+	r.Filters = append(r.Filters,
+		[]*eskip.Filter{
+			{
+				// flowId("reuse")
+				Name: filters.FlowIdName,
+				Args: []interface{}{
+					flowid.ReuseParameterValue,
+				},
+			},
+			{
+				// forwardToken("X-TokenInfo-Forward", "uid", "scope", "realm")
+				Name: filters.ForwardTokenName,
+				Args: []interface{}{
+					// TODO(sszuecs): in the future should be configurable
+					"X-TokenInfo-Forward",
+					"uid",
+					"scope",
+					"realm",
+				},
+			},
+		}...)
+
+	// optional cors
+	if len(allowedOrigins) > 0 {
+		r.Filters = append(r.Filters,
+			&eskip.Filter{
+				// corsOrigin("https://foo.example.org", "https://bar.example.com")
+				Name: filters.CorsOriginName,
+				Args: allowedOrigins,
+			},
+		)
+	}
+
+	return r
+}
+
+func createRatelimitRoutes(r *eskip.Route, m *FabricMethod, name, path string) []*eskip.Route {
+	routes := make([]*eskip.Route, 0, len(m.Ratelimit.Target))
+
+	for i, rTarget := range m.Ratelimit.Target {
+		rr := eskip.Copy(r)
+		rr.Id = fmt.Sprintf("%s%d", rr.Id, i)
+
+		// add predicate to match client application
+		rr.Predicates = append(rr.Predicates,
+			&eskip.Predicate{
+				Name: predicates.JWTPayloadAllKVName,
+				Args: []interface{}{
+					"sub", // TODO(sszuecs) maybe configurable in the future
+					rTarget.UID,
+				},
+			},
+		)
+
+		// find and replace ratelimit: type, group, rate. period stays the same
+		for j := range rr.Filters {
+			if rr.Filters[j].Name == filters.ClusterClientRatelimitName {
+				// replace clusterClientRatelimit with clusterRatelimit,
+				// because we have separate routes per UID and we can scale
+				// shards with clusterRatelimit
+				rr.Filters[j].Name = filters.ClusterRatelimitName
```

Comment:
the filter was changed from ClusterClientRatelimitName to ClusterRatelimitName, because we already know the client by applying JWTPayloadAllKVName predicate with `sub, rTarget.UID`